domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
 # Private Methods
-##################################################################################################
+###############################################################################
 
     def _set_at_home(self):
+        """Update home status to at home and debug LEDs (if enabled)."""
+        self._turn_led_on(leds=['input_2'])
         self._at_home = True
 
     def _set_not_home(self):
+        """Update home status to not at home and debug LEDs (if enabled)."""
+        self._turn_led_off(leds=['input_2'])
         self._at_home = False
 
-    def _increment_count(self, device=None):
-        print(f"{device} activated _increment_count")
+    def _increment_count(self):
+        """
+        Private method used for callback function of the encoder DOD.
+
+        Calling this method will toggle the encoder debug LED (if enabled)
+        and increment or decrement the encoder_count instance variable,
+        depending on the current rotation direction of the dome.
+
+        If the current dome direction cannot be determined, the last recorded
+        direction is adopted.
+        """
+        print(f"Encoder activated _increment_count")
+        self._turn_led_on(leds=['input_1'])
+        time.sleep(0.01)
+        self._turn_led_off(leds=['input_1'])
         if self.current_direction == "CW":
             self.encoder_count += 1
         elif self.current_direction == "CCW":
             self.encoder_count -= 1
+        # I'm unsure if this is the best way to handle a situation like this
         elif self.current_direction is None:
             if self.last_direction == "CW":
                 self.encoder_count += 1
             elif self.last_direction == "CCW":
                 self.encoder_count -= 1
 
-    def _move_west(self, degrees_west):
-        print(f"Moving {degrees_west} West")
-        if self._west_is_ccw:
-            cmd_status = self.move_ccw(degrees_west)
-        else:
-            cmd_status = self.move_cw(degrees_west)
-
-        return cmd_status
-
-    def _move_cw(self, degrees):
-        print(f"Sending GPIO move_cw({degrees}) command.")
+    def _az_to_ticks(self, az):
+        """Convert degrees (azimuth) to equivalent in encoder tick count."""
+        return az / self.az_per_tick
+
+    def _ticks_to_az(self, ticks):
+        """Convert encoder tick count to equivalent in degrees (azimuth)."""
+        return ticks * self.az_per_tick
+
+    def _move_cw(self):
+        """Set dome to move clockwise."""
+        # if testing, deactivate the home_sernsor_pin to simulate leaving home
+        if self.testing and self._at_home:
+            self.home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CW
         self.current_direction = "CW"
+        # set the direction relay switch to CW position
+        self.direction_relay.on()
```

Comment:
```suggestion
        self.direction_CW_relay.on()
```
Turning on() just a `direction_relay` is less clear than turning off/on a `direction_CW_relay`. Right?